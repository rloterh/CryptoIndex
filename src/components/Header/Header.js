import React from 'react';
import { Navbar, Button, Nav } from 'react-bootstrap';
import { NavLink } from 'react-router-dom';
import './Header.scss';

const Header = () => (
  <>
    <header>
      <Navbar bg="light" variant="light">
        <Nav.Link as={NavLink} to="/">
          <Button id="back-arrow">
            <i className="fas fa-arrow-left" />
          </Button>
        </Nav.Link>
        <Nav.Link as={NavLink} to="/">
          <Navbar.Brand>Crypto Index</Navbar.Brand>
        </Nav.Link>
        <Button id="menu-settings">
          <i className="fas fa-cog" />
        </Button>
      </Navbar>
    </header>
  </>
);

export default Header;
