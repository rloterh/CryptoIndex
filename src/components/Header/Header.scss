
.navbar {
  background-color: #dd9822 !important;
  justify-content: space-around;
  font-family: 'Roboto', sans-serif;
}

.navbar-brand {
  color: #fff !important;
  font-weight: bold;
  font-size: 1.8rem;
  letter-spacing: 1.8px;
}
