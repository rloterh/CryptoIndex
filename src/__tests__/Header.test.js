import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from '../components/Header/Header';

describe('Header', () => {
  test('Header component render correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Header />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
