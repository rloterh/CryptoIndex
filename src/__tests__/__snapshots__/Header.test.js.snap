// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header component render correctly 1`] = `
<header>
  <nav
    className="navbar navbar-expand navbar-light bg-light"
    data-testid="nav-bar"
  >
    <a
      aria-current="page"
      className="nav-link active"
      disabled={false}
      href="/"
      onClick={[Function]}
    >
      <button
        className="btn btn-primary"
        disabled={false}
        id="back-arrow"
        type="button"
      >
        <i
          className="fas fa-arrow-left"
        />
      </button>
    </a>
    <a
      aria-current="page"
      className="nav-link active"
      disabled={false}
      href="/"
      onClick={[Function]}
    >
      <span
        className="navbar-brand"
      >
        Crypto Index
      </span>
    </a>
    <button
      className="btn btn-primary"
      disabled={false}
      id="menu-settings"
      type="button"
    >
      <i
        className="fas fa-cog"
      />
    </button>
  </nav>
</header>
`;
